sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -258,9 +258,9 @@ def _check_categories(self, X):
         if not np.any(is_categorical):
             return None, None
 
-        # compute the known categories in the training data. We need to do
-        # that here instead of in the BinMapper because in case of early
-        # stopping, the mapper only gets a fraction of the training data.
+        # Compute the known categories in the training data. We need to do that
+        # here instead of in the BinMapper because in case of early stopping,
+        # the mapper only gets a fraction of the training data.
```

Comment:
The "because in case of" tripped me up the first time I read this. Not sure if "In the case of early stopping we have to do this here instead of in BinMapper because the mapper only gets ..." is better, but maybe.

Reply:
I got the same feeling when I read this sentence the first time :) I will push your suggestion.

Reply:
I pushed another rewording in this end :)